mod/groups/actions/groups/edit.php

```diff
@@ -52,32 +61,30 @@ function profile_array_decoder(&$v) {
 }
 
 // Assume we can edit or this is a new group
-if (sizeof($input) > 0) {
-	foreach($input as $shortname => $value) {
-		// update access collection name if group name changes
-		if (!$is_new_group && $shortname == 'name' && $value != $group->name) {
-			$group_name = html_entity_decode($value, ENT_QUOTES, 'UTF-8');
-			$ac_name = sanitize_string(elgg_echo('groups:group') . ": " . $group_name);
-			$acl = get_access_collection($group->group_acl);
-			if ($acl) {
-				// @todo Elgg api does not support updating access collection name
-				$db_prefix = elgg_get_config('dbprefix');
-				$query = "UPDATE {$db_prefix}access_collections SET name = '$ac_name'
-					WHERE id = $group->group_acl";
-				update_data($query);
-			}
-		}
-
-		if ($value === '') {
-			// The group profile displays all profile fields that have a value.
-			// We don't want to display fields with empty string value, so we
-			// remove the metadata completely.
-			$group->deleteMetadata($shortname);
-			continue;
+foreach ($input as $shortname => $value) {
+	// update access collection name if group name changes
+	if (!$is_new_group && $shortname == 'name' && $value != $group->name) {
+		$group_name = html_entity_decode($value, ENT_QUOTES, 'UTF-8');
+		$ac_name = sanitize_string(elgg_echo('groups:group') . ": " . $group_name);
+		$acl = get_access_collection($group->group_acl);
+		if ($acl) {
+			// @todo Elgg api does not support updating access collection name
+			$db_prefix = elgg_get_config('dbprefix');
+			$query = "UPDATE {$db_prefix}access_collections SET name = '$ac_name'
+				WHERE id = $group->group_acl";
+			update_data($query);
 		}
+	}
 
-		$group->$shortname = $value;
+	if ($value === '' && !in_array($shortname, ['name', 'description'])) {
```

Comment:
Technically this was a harmless bug that might delete metadata called "name" or "description".